src/enterdb_lib.erl

```diff
@@ -1498,7 +1498,7 @@ apply_update_op(Op, DBValue, #{data_model := DataModel,
 	{ok, DBValue} ->
 	    {ok, DBValue, []};
 	{ok, UpdatedDBValue} ->
-	    IndexTerms = get_index_terms(Mapper, IndexOn, Columns),
+	    IndexTerms = get_index_terms(Mapper, IndexOn, UpdatedColumns),
```

Comment:
Here it would not help since UpdatedColumns is used but this bug made me think if it would be good to use warnings as errors compile flag. What do you think? Are there too many warnings to resolve?